hazelcast/src/main/java/com/hazelcast/client/impl/clientside/ClientDiscoveryServiceBuilder.java

```diff
@@ -87,6 +85,11 @@ public ClientDiscoveryService build() {
             ClientNetworkConfig networkConfig = config.getNetworkConfig();
             SocketInterceptor interceptor = initSocketInterceptor(networkConfig.getSocketInterceptorConfig());
             ICredentialsFactory credentialsFactory = initCredentialsFactory(config);
+            if (credentialsFactory == null) {
+                credentialsFactory = new StaticCredentialsFactory(new UsernamePasswordCredentials(config.getClientName(), ""));
+            }
+            credentialsFactory.configure(cs -> {
+            });
```

Comment:
The `configure()` should be always called for factory implementation. There is no specific callback supported currently on the client-side so such an empty handler should be OK, but probably throwing the `UnsupportedCallbackException` will work better here (instead of the empty statement).